lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -50,6 +50,9 @@ def compute(self, student, course_structure, scores_client, submissions_scores):
         ):
             self._compute_block_score(student, descendant_key, course_structure, scores_client, submissions_scores)
 
+        # self.scores may hold outdated data, force it to refresh on next access
+        lazy.invalidate(self, 'scores')
```

Comment:
Since there are separate calls to `_compute_block_score` for each descendant, could you end up in a situation where some have succeeded and some have failed? You would want to invalidate `scores` even if there is an error and this line is not reached, right?